reposerver/repository/repository.go

```diff
@@ -1253,6 +1349,298 @@ func getRepoCredential(repoCredentials []*v1alpha1.RepoCreds, repoURL string) *v
 	return nil
 }
 
+func copyFilesystemSubtree(srcRoot string, srcPath string, dstRoot string, dstPath string, hardfail bool) error {
```

Comment:
This is a pretty huge function. Can we break it down into smaller chunks? We can maybe modularize the code of copying files and reuse the function for directories as well. WDYT?

Reply:
I've moved the walking lambda function out, and refactored the logging statements, so it's not so huge. I'm not sure about going further as I don't really know what parts and how are to be reused. Might be better to do that when it's needed, like how I've done it with the OOB checks.